metashare/api/jobs.py

```diff
@@ -192,6 +205,17 @@ def create_branches_on_github_then_create_scratch_org(
 )
 
 
+def check_if_changes_on_org(*, scratch_org, user):
+    with report_errors_on_check_changes(scratch_org):
+        scratch_org.has_changes = sf_org_changes.sf_org_has_changes(
+            scratch_org=scratch_org, user=user
+        )
+        scratch_org.save()
```

Comment:
This should use `mark_refreshing_changes` so that it sets `currently_refreshing_changes` and triggers WS events accordingly?

I'll note that I'm not convinced abstracting the `mark_refreshing_changes` logic is actually helpful, nor using a tracker on `currently_refreshing_changes` to trigger the `SCRATCH_ORG_UPDATE` event. Aside from the bug we discovered with the job-instantiated tracker not having the same context as the model-instantiated one, it seems this amount of abstraction will result in multiple unnecessary `scratch_org.save()` calls?

As I understand it, this is what we want:

- On every org `GET`, set `currently_refreshing_changes: True` in the model (and in the `GET` response), and trigger a job to check for changes.
- On success, set `currently_refreshing_changes: False` and `scratch_org.changes` to the changes. Trigger a `SCRATCH_ORG_UPDATE` event.
- On failure, set `currently_refreshing_changes: False` and `scratch_org.changes: None`. Trigger either a `SCRATCH_ORG_UPDATE` event or `SCRATCH_ORG_FETCH_CHANGES_FAILED` event.

Reply:
Also: I think we only want to do this for Dev orgs, not QA orgs.